### PR TITLE
agent: add tests for create_logger_task function

### PR DESCRIPTION
**Pull request by bradenrayhorn**
Add tests for create_logger_task function in src/main.rs.

Fixes: #4113

Signed-off-by: Braden Rayhorn <bradenrayhorn@fastmail.com>

I am interested in feedback on ideas for being able to test more with this function. One particular problem I ran into when trying to add more tests was getting the `util::get_vsock_stream` future used in `create_logger_task` to complete execution.

### CODE REVIEW

File: src/agent/src/main.rs [hunk 1]
```diff
@@ -416,3 +416,59 @@ fn reset_sigpipe() {
...
+            defer!({
+                // rfd is closed by the crate_logger_task function, but we will attempt
+                // to close in case of a failure
+                let _ = unistd::close(rfd);
+                unistd::close(wfd).unwrap();
+            });
+
+            let (shutdown_tx, shutdown_rx) = channel(true);
+
+            shutdown_tx.send(true).unwrap();
```
Comment (liubin) [inline src/agent/src/main.rs:467]: Is it better moving `shutdown_tx.send(true).unwrap();` after line 468?

File: src/agent/src/main.rs [hunk 2]
```diff
@@ -416,3 +416,59 @@ fn reset_sigpipe() {
...
+            if d.test_user == TestUserType::RootOnly {
+                skip_if_not_root!();
+            } else if d.test_user == TestUserType::NonRootOnly {
+                skip_if_root!();
+            }
+
+            let msg = format!("test[{}]: {:?}", i, d);
+            let (rfd, wfd) = unistd::pipe2(OFlag::O_CLOEXEC).unwrap();
+            defer!({
+                // rfd is closed by the crate_logger_task function, but we will attempt
```
Comment (liubin) [inline src/agent/src/main.rs:459]: Precisely say is that rfd will be closed by PipeStream?

File: src/agent/src/main.rs (hunk 1 above)
Comment (bradenrayhorn) [reply src/agent/src/main.rs:467]: I'm open to suggestions on this, but with the way the test is setup now, I don't think we can. If it is called after line 468, the `create_logger_task.await` will be waiting indefinitely. Sending the `shutdown` call makes sure the `create_logger_task.await` will finish.

**Review by jodh-intel — APPROVED** [review]
Thanks @bradenrayhorn.

lgtm

**Comment by jodh-intel** [comment]
Hmm - some of the CI's are failing with http 503 (service unavailable):

```
docker.io/library/nginx:1.15-alpine: resolving      |[32m[0m--------------------------------------| 
elapsed: 0.1 s                       total:   0.0 B (0.0 B/s)                                         
time="2022-04-25T07:05:38-04:00" level=info msg="trying next host" error="failed to authorize: failed to fetch anonymous token: unexpected status: 503 Service Unavailable" host=registry-1.docker.io
ctr: failed to resolve reference "docker.io/library/nginx:1.15-alpine": failed to authorize: failed to fetch anonymous token: unexpected status: 503 Service Unavailable
ERROR: Unable to retry docker image docker.io/library/nginx:1.15-alpine
```

```
INFO: Build cloud-hypervisor enabling the following features: tdx
Error response from daemon: Head "https://registry-1.docker.io/v2/cloudhypervisor/dev/manifests/20220405-0": received unexpected HTTP status: 503 Service Unavailable
Sending build context to Docker daemon  6.144kB

Step 1/24 : FROM ubuntu:20.04 as dev
Head "https://registry-1.docker.io/v2/library/ubuntu/manifests/20.04": received unexpected HTTP status: 503 Service Unavailable
[Cloud Hypervisor] Error pulling/building container image. Aborting.
```

We also seem to have hit a golang unit test bug randomly. I've raised https://github.com/kata-containers/kata-containers/issues/4150, but will now restart that CI since it's clearly unreleated to this change...

**Comment by jodh-intel** [comment]
Ping for one more review y'all!

**Review by liubin — APPROVED** [review]

**Pull request merged**